packages/@aws-cdk/aws-batch/README.md

```diff
@@ -13,39 +14,271 @@
+    // Pay full-price for all provisioned resources
+    type: batch.ComputeResourceType.ON_DEMAND,
+
+    // Uses AWS recommended AMIs for GPU-based workloads
+    image: new ecs.EcsOptimizedAmi({
+      generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2,
+      hardwareType: ecs.AmiHardwareType.GPU,
+    }),
+
+    // 'ec2KeyPair' specifies the name of the related EC2 KeyPair for SSH access
+    ec2KeyPair: 'dev',
+
+    // 'instanceRole' defines the privileges the job will have when running
+    instanceRole: new iam.Role(stack, 'compute-env-instance-role', {
+      assumedBy: new iam.ServicePrincipal('batch.amazonaws.com'),
+    }),
+
+    // Only pick `p3.2xlarge` or `p3.8xlarge` instances to provision
+    instanceTypes: [
+      ec2.InstanceType.of(ec2.InstanceClass.P3, ec2.InstanceSize.XLARGE2),
+      ec2.InstanceType.of(ec2.InstanceClass.P3, ec2.InstanceSize.XLARGE8),
+    ],
+
+    // At most, provision up to 40 vCPUs worth of instances. For example, 5x p3.8xlarge => 40 vCPU
+    maxvCpus: 40,
+
+    // 'minvCpus' specifies our required minimal vCPU availability when no jobs exists
+    minvCpus: 8,
+
+    // 'vpc' and 'vpcSubnets' locks down our containers to the private subnet of our VPC
+    vpc,
+    vpcSubnets: {
+      subnetType: ec2.SubnetType.PRIVATE,
+    },
+  },
+});
+```
 
-### Usage
+### Launch template support
+
+#### Usage
 Simply define your Launch Template:
 ```typescript
-    const myLaunchTemplate = new ec2.CfnLaunchTemplate(this, 'LaunchTemplate', {
-      launchTemplateName: 'extra-storage-template',
-      launchTemplateData: {
-        blockDeviceMappings: [
-          {
-            deviceName: '/dev/xvdcz',
-            ebs: {
-              encrypted: true,
-              volumeSize: 100,
-              volumeType: 'gp2'
-            }
-          }
-        ]
+const myLaunchTemplate = new ec2.CfnLaunchTemplate(this, 'LaunchTemplate', {
+  launchTemplateName: 'extra-storage-template',
+  launchTemplateData: {
+    blockDeviceMappings: [
+      {
+        deviceName: '/dev/xvdcz',
+        ebs: {
+          encrypted: true,
+          volumeSize: 100,
+          volumeType: 'gp2'
+        }
       }
-    });
+    ]
+  }
+});
 ```
+
 and use it:
 
 ```typescript
-    const myComputeEnv = new batch.ComputeEnvironment(this, 'ComputeEnv', {
-      computeResources: {
-        launchTemplate: {
-          launchTemplateName: myLaunchTemplate.launchTemplateName as string, //or simply use an existing template name
-        },
-        vpc,
-      },
-      computeEnvironmentName: 'MyStorageCapableComputeEnvironment',
-    });
+const myComputeEnv = new batch.ComputeEnvironment(this, 'ComputeEnv', {
+  computeResources: {
+    launchTemplate: {
+      launchTemplateName: myLaunchTemplate.launchTemplateName as string, //or simply use an existing template name
+    },
+    vpc,
+  },
+  computeEnvironmentName: 'MyStorageCapableComputeEnvironment',
+});
+```
+
+### Handling unexpected peaks using Spot-Based resources
```

Comment:
Based on the comments, I removed this second to avoid confusion. It was more of a scenario building section to help others understanding. But I see best that we shouldn't presume.